test/spec/data/serialized-data-service.spec.js

```diff
@@ -0,0 +1,33 @@
+const { SerializedDataService } = require("mod/data/service/serialized-data-service.mod/serialized-data-service");
+const { defaultEventManager } = require("mod/core/event/event-manager");
+const { DataService } = require("mod/data/service/data-service");
+
+const Country = require("mod/data/model/country").montageObject;
+
+describe("SerializedDataService", () => {
+    let serializedDataService;
+    let mainService;
+
+    beforeEach(async () => {
+        mainService = new DataService();
+        serializedDataService = new SerializedDataService();
+        // defaultEventManager.application.mainService = mainService;
+
+        // Register some instances for testing
+        serializedDataService.registerInstancesLocationForType(Country, "test/spec/data/logic/model/countries.mjson");
+
+        // FIXME: can't register for an array of types, an error is thrown
+        // await mainService.registerChildService(serializedDataService, [Country]);
+
+        await mainService.registerChildService(serializedDataService, Country);
```

Comment:
@marchant @tejaede What am I missing here?

Reply:
I don't know, are you referring to handleReadOperation() not being called in SerializedDataService.js?

Have you tried to debug it, following the code to see what happens? Because

const Country = require("mod/data/model/country").montageObject;

seems wrong - this is a JS file, so there's no "montageObject" symbol on the exports, so you might very well passing nothing?